### PR TITLE
Kill node process on exit

### PR DESCRIPTION
**Pull request by yhaspel**

### CODE REVIEW

File: desktop/nodeManager.js [hunk 1]
```diff
@@ -3,13 +3,26 @@ import os from 'os';
...
 
 const osTargetNames = {
   Darwin: 'mac',
   Linux: 'linux',
   Windows_NT: 'windows'
 };
 
+const getPidByName = ({ name }) => {
+  return new Promise((resolve, reject) => {
+    try {
```
Comment (IlyaVi) [inline desktop/nodeManager.js]: there are 2 options to write this function:
`find .... then(), (err) { reject(); }`
or
`try {
   const list = await find
   ...
}
 catch {}
`
but not mixing two ways to deal async response

File: desktop/nodeManager.js [hunk 2]
```diff
@@ -3,13 +3,26 @@ import os from 'os';
...
+      find('name', name).then((list) => {
+        resolve(list?.length && list[0].pid ? list[0].pid : null);
```
Comment (IlyaVi) [inline desktop/nodeManager.js]: valid return is always array and thus it always has length prop

File: desktop/nodeManager.js [hunk 3]
```diff
@@ -30,6 +43,24 @@ class NodeManager {
   };
 
   static hardRefresh = ({ browserWindow }) => browserWindow.reload();
+
+  static killNodeProcess = async ({ event }) => {
+    try {
+      const isDevMode = process.env.NODE_ENV === 'development' || process.env.DEBUG_PROD === 'true';
```
Comment (IlyaVi) [inline desktop/nodeManager.js]: DEBUG_PROD means that the app is packed as production app so it won't have miniMesh.sh start point

File: desktop/nodeManager.js [hunk 4]
```diff
@@ -30,6 +43,24 @@ class NodeManager {
...
+      if (isDevMode) {
```
Comment (IlyaVi) [inline desktop/nodeManager.js]: add TODO to remove this immediately when dev mode will use node binary too

File: app/infra/nodeService/nodeService.js [hunk 5]
```diff
@@ -21,6 +21,20 @@ class NodeService {
   static hardRefresh() {
     ipcRenderer.send(ipcConsts.HARD_REFRESH);
   }
+
+  static killNodeProcess() {
+    ipcRenderer.send(ipcConsts.QUIT_NODE);
+    return new Promise<string, Error>((resolve: Function, reject: Function) => {
```
Comment (IlyaVi) [inline app/infra/nodeService/nodeService.js]: it is meaningless to wait for response of kill call especially when error is ignored

File: desktop/nodeManager.js (hunk 2 above)
Comment (yhaspel) [reply desktop/nodeManager.js]: True, but I want to know if the length is larger than 0 as well.

Comment (IlyaVi) [reply desktop/nodeManager.js]: Ok, but '?' is useless here

Comment (yhaspel) [reply desktop/nodeManager.js]: true

File: app/infra/nodeService/nodeService.js (hunk 5 above)
Comment (yhaspel) [reply app/infra/nodeService/nodeService.js]: sendSync sends and closes the spacemesh node running, but app is stuck and not closing. await in event listeners file also not working. 
I'm keeping the promise for now and will try to look for another solution.

File: app/components/common/QuitDialog.js [hunk 6]
```diff
@@ -6,6 +6,7 @@ import { CorneredWrapper, Button } from '/basicComponents';
 import { smColors, ipcConsts, nodeConsts } from '/vars';
 import styled from 'styled-components';
 import { notificationsService } from '/infra/notificationsService';
+// import { nodeService } from '/infra/nodeService';
```
Comment (IlyaVi) [inline app/components/common/QuitDialog.js]: remove

File: desktop/nodeManager.js [hunk 7]
```diff
@@ -1,22 +1,46 @@
+/* eslint-disable no-param-reassign */
```
Comment (IlyaVi) [inline desktop/nodeManager.js]: move to specific lines, bad pattern - can hide bugs

File: desktop/nodeManager.js [hunk 8]
```diff
@@ -1,22 +1,46 @@
...
 const child = require('child_process').execFile;
+const find = require('find-process');
 
 const osTargetNames = {
   Darwin: 'mac',
   Linux: 'linux',
   Windows_NT: 'windows'
 };
 
+// const getPidByName = async ({ name }) => {
```
Comment (IlyaVi) [inline desktop/nodeManager.js]: remove

File: desktop/nodeManager.js [hunk 9]
```diff
@@ -1,22 +1,46 @@
...
+//   } catch {
+//     return null;
+//   }
+// };
+
+const getPidByName = async ({ name }) => {
+  return find('name', name).then((list) => {
+    return list.length && list[0].pid ? list[0].pid : null;
+  });
+  // try {
```
Comment (IlyaVi) [inline desktop/nodeManager.js]: remove

File: desktop/nodeManager.js [hunk 10]
```diff
@@ -1,22 +1,46 @@
...
+// const getPidByName = async ({ name }) => {
+//   try {
+//     const list = await find('name', name);
+//     return list.length && list[0].pid ? list[0].pid : null;
+//   } catch {
+//     return null;
+//   }
+// };
+
+const getPidByName = async ({ name }) => {
```
Comment (IlyaVi) [inline desktop/nodeManager.js]: remove async

**Pull request merged**